src/encoder.rs

```diff
@@ -182,8 +182,12 @@ pub struct Sequence {
 
 impl Sequence {
   pub fn new(config: &EncoderConfig) -> Sequence {
-    let width_bits = 32 - (config.width as u32).leading_zeros();
-    let height_bits = 32 - (config.height as u32).leading_zeros();
+    let max_width =
+      if config.max_width > 0 { config.max_width } else { config.width };
+    let max_height =
+      if config.max_height > 0 { config.max_height } else { config.height };
+    let width_bits = 32 - (max_width as u32).leading_zeros();
+    let height_bits = 32 - (max_height as u32).leading_zeros();
```

Comment:
Fix this while at it by making it max_{width,height} - 1, same as it was done in 325ae51558cec45e7c7e96b0121c3b966cdeca7c, so once you rebase this PR to solve the conflicts in src/header.rs you don't introduce a regression.